miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/AttachmentController.java

```diff
@@ -0,0 +1,97 @@
+package uk.ac.bbsrc.tgac.miso.webapp.controller;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.acls.model.NotFoundException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.MultipartHttpServletRequest;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Attachable;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.FileAttachment;
+import uk.ac.bbsrc.tgac.miso.service.FileAttachmentService;
+
+@Controller
+@RequestMapping("/attachments")
+public class AttachmentController {
+
+  @Autowired
+  private FileAttachmentService fileAttachmentService;
+
+  @Value("${miso.fileStorageDirectory}")
+  private String fileStorageDirectory;
+
+  @PostMapping(value = "/{entityType}/{entityId}")
+  @ResponseStatus(HttpStatus.OK)
+  public void acceptUpload(@PathVariable String entityType, @PathVariable long entityId, MultipartHttpServletRequest request)
+      throws IllegalStateException, IOException {
+    Attachable item = fileAttachmentService.get(entityType, entityId);
+    if (item == null) {
+      throw new NotFoundException(entityType + " not found");
+    }
+
+    for (MultipartFile fileItem : getMultipartFiles(request)) {
+      fileAttachmentService.add(item, fileItem);
+    }
+  }
+
+  private List<MultipartFile> getMultipartFiles(MultipartHttpServletRequest request) {
+    List<MultipartFile> files = new ArrayList<>();
+    Map<String, MultipartFile> fMap = request.getFileMap();
+    for (String fileName : fMap.keySet()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2864)